src/Controller/FindologicConfigController.php

```diff
@@ -65,8 +67,16 @@ public function saveConfiguration(Request $request): Response
      */
     public function batchSaveConfiguration(Request $request): Response
     {
+        $shopkeys = [];
         foreach ($request->request->all() as $key => $config) {
             [$salesChannelId, $languageId] = explode('-', $key);
+
+            $currentShopkey = $config['FinSearch.config.shopkey'];
+            if ($currentShopkey && in_array($currentShopkey, $shopkeys, false)) {
+                throw new ShopkeyAlreadyExistsException();
+            }
+
+            $shopkeys[] = $currentShopkey;
```

Comment:
This isn't being used anywhere.

Reply:
Oh nevermind it actually is used, but maybe this could use some refactoring, as it really is kinda hard to read.